### DISCUSSION TITLE
Using an isolated RS485 board without TALK PIN

### DISCUSSION THREAD

**Discussion by Sleeper85** [txubelaxu/esphome-jk-bms#19] [Ideas]
Hi @txubelaxu 

Could we use another RS485 board as I propose in my [README](https://github.com/Sleeper85/esphome-jk-bms-can/blob/development/documents/README/BMS_JK-PB_RS485_solution.md) and have the option of not having to define a GPIO for the `TALK PIN` ?

This is the [RS485 board (high speed dual)](https://a.aliexpress.com/_EueIZT5) used with the [PVbrain2 PCB](https://github.com/SeByDocKy/pvbrain2).

**All the GPIOs** of the ESP32-S3 are already used for different functions and distributed on the PCB.
So I would like to be able to not have to define the `TALK PIN`.

Thanks for your help and development of this component.

**Comment by txubelaxu** [comment]
I'll try to change the code to "optional" talk_pin

**Comment by txubelaxu** [comment]
Hi @Sleeper85,

I have just change that setting to OPTIONAL, so you should remove from yaml, because it is not mandatory now.

Could you test, please? Tell me, please, if you find any problem.

Thanks!!!

**Comment by Sleeper85** [comment]
Thanks, I confirm that I can compile this new code without problems but I cannot test it personally because I don't have a JK-PB BMS.

@txubelaxu would you be interested in testing it yourself ?

Otherwise, is ( @widget4145 / @cinusik / @virus100b ) willing to test this RS485 board ?

The goal is to be able to confirm that this RS485 board (without talk pin) works well.
It is a very good isolated RS485 board and I would like to be able to recommend it.

[RS485 isolated board (choose this version => high speed dual)](https://a.aliexpress.com/_EueIZT5)

![ADUM3201_B0505XT_TTL_to_RS485_isolation_high_speed_dual_board](https://github.com/user-attachments/assets/43c5d67f-698d-41e0-831a-934d7e3e1afd)

**Comment by txubelaxu** [comment]
Yes, Iam compiling esp-jk-bms-can now and I will test it.

**Comment by cinusik** [comment]
@Sleeper85 As soon as I will have some time will test it with the RS485 Atomic board from M5, as I don't have one mentioned but it should also work. Let you guys know once tested.

**Reply by ecornwell01** [reply]
I've got a different board without a talk pin as well that I'll hopefully be testing soon.  I'm compiling now.
